packages/table/src/headers/columnHeader.tsx

```diff
@@ -43,102 +37,67 @@ export interface IColumnHeaderProps extends
-    };
-
-    private activationCol: number;
-
-    public componentDidMount() {
-        if (this.props.selectedRegions != null && this.props.selectedRegions.length > 0) {
-            // we already have a selection defined, so we'll want to enable reordering interactions
-            // right away if other criteria are satisfied too.
-            this.setState({ hasSelectionEnded: true });
-        }
-    }
-
-    public componentWillReceiveProps(nextProps?: IColumnHeaderProps) {
-        if (nextProps.selectedRegions != null && nextProps.selectedRegions.length > 0) {
-            this.setState({ hasSelectionEnded: true });
-        } else {
-            this.setState({ hasSelectionEnded: false });
-        }
+    public render() {
+        return (
+            <AbstractHeader
+                allowMultipleSelection={this.props.allowMultipleSelection}
+                convertPointToIndex={this.convertPointToColumn}
+                endIndex={this.props.columnIndexEnd}
+                fullRegionCardinality={RegionCardinality.FULL_COLUMNS}
+                getCellExtremaClasses={this.getCellExtremaClasses}
+                getCellIndexClass={Classes.columnCellIndexClass}
+                getCellSize={this.getColumnWidth}
+                getDragCoordinate={this.getDragCoordinate}
+                getIndexClass={Classes.columnIndexClass}
+                getMouseCoordinate={this.getMouseCoordinate}
+                grid={this.props.grid}
+                handleResizeDoubleClick={this.handleResizeDoubleClick}
+                handleResizeEnd={this.handleResizeEnd}
+                handleSizeChanged={this.handleSizeChanged}
+                headerCellIsReorderablePropName={"isColumnReorderable"}
+                headerCellIsSelectedPropName={"isColumnSelected"}
+                isCellSelected={this.isCellSelected}
+                isGhostIndex={this.isGhostIndex}
+                isReorderable={this.props.isReorderable}
+                isResizable={this.props.isResizable}
+                loading={this.props.loading}
+                locator={this.props.locator}
+                maxSize={this.props.maxColumnWidth}
+                minSize={this.props.minColumnWidth}
+                onFocus={this.props.onFocus}
+                onLayoutLock={this.props.onLayoutLock}
+                onReordered={this.props.onReordered}
+                onReordering={this.props.onReordering}
+                onResizeGuide={this.props.onResizeGuide}
+                onSelection={this.props.onSelection}
+                renderGhostCell={this.renderGhostCell}
+                renderHeaderCell={this.props.cellRenderer}
+                resizeOrientation={Orientation.VERTICAL}
+                selectedRegions={this.props.selectedRegions}
+                selectedRegionTransform={this.props.selectedRegionTransform}
+                startIndex={this.props.columnIndexStart}
+                toRegion={this.toRegion}
+                viewportRect={this.props.viewportRect}
```

Comment:
Not all props should be passed down (in particular, isColumnReorderable and isColumnSelected, whose names are specific to this component). However, can do something like:

const { isColumnReorderable, isColumnSelected, ...spreadableProps } = this.props;
return <AbstractHeader {...this.spreadableProps} ...etc...  />;